backend/tests/unit/test_agent_outdated_detection.py

```
@@ -0,0 +1,436 @@
+
+        # Should not raise (will return 0 since no user prefs enabled)
+        result = svc.notify_agent_status(
+            team_id=test_team.id,
+            agent=agent,
+            transition_type="agent_outdated",
+        )
+        assert isinstance(result, int)
+
+    @patch("backend.src.services.notification_service.NotificationService")
+    def test_heartbeat_triggers_outdated_notification(
+        self, mock_ns_class, test_db_session, agent_service, agent,
```

Ruff (0.15.1)

[warning] Unused method argument: `mock_ns_class`

(ARG002)